server/api.go

```diff
@@ -595,3 +599,56 @@ func (p *Plugin) updateSettings(c *UserContext, w http.ResponseWriter, r *http.R
 
 	p.writeAPIResponse(w, info.Settings)
 }
+
+type SubscriptionResponse struct {
+	RepositoryName string   `json:"repository_name"`
+	RepositoryURL  string   `json:"repository_url"`
+	Features       []string `json:"features"`
+	CreatorID      string   `json:"creator_id"`
+}
+
+func subscriptionsToResponse(config *configuration, subscriptions []*subscription.Subscription) []SubscriptionResponse {
+	gitlabURL, _ := url.Parse(config.GitlabURL)
+
+	subscriptionResponses := make([]SubscriptionResponse, 0, len(subscriptions))
+
+	for _, subscription := range subscriptions {
+		features := []string{}
+		if len(subscription.Features) > 0 {
+			features = strings.Split(subscription.Features, ",")
+		}
+
+		subscriptionResponses = append(subscriptionResponses, SubscriptionResponse{
+			RepositoryName: subscription.Repository,
+			RepositoryURL:  gitlabURL.JoinPath(subscription.Repository).String(),
+			Features:       features,
+			CreatorID:      subscription.CreatorID,
+		})
+	}
+
+	return subscriptionResponses
+}
+
+func (p *Plugin) getChannelSubscriptions(c *UserContext, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	channelID := vars["channel_id"]
+
+	if !p.API.HasPermissionToChannel(c.UserID, channelID, model.PermissionReadChannel) {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Not authorized.", StatusCode: http.StatusUnauthorized})
+		return
+	}
+
+	config := p.getConfiguration()
+	subscriptions, err := p.GetSubscriptionsByChannel(channelID)
+	if err != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Unable to get subscriptions by channel.", StatusCode: http.StatusInternalServerError})
+		return
+	}
+
+	resp := subscriptionsToResponse(config, subscriptions)
+
+	b, _ := json.Marshal(resp)
```

Comment:
We should probably check the err returned here. IIRC, we've had some incidents on the server around JSON Marshalling so a quick error check helps.